### PR TITLE
Session where

### PR DESCRIPTION
**Pull request by g-krisztian**
### Description
We had some cases where it's handy to have a predefined filtering on session-dump

### Tester info
Component tests implemented for db-backend, unit tests for in-memory backend

### CODE REVIEW

**Comment by jehaby** [comment]
could you please provide some examples of when this feature is necessary? i suspect it might look strange with e.g. redis session backend

File: test/resources/migrations/20211227102201-session.up.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE sessions (
+    session_data json not null,
```
Comment (jehaby) [inline test/resources/migrations/20211227102201-session.up.sql:2]: is there a reason to you json instead of jsonb here?

**Comment by g-krisztian** [comment]
> could you please provide some examples of when this feature is necessary? i suspect it might look strange with e.g. redis session backend

for example, fetching all sessions for given user

File: test/resources/migrations/20211227102201-session.up.sql (hunk 1 above)
Comment (g-krisztian) [reply test/resources/migrations/20211227102201-session.up.sql:2]: nope

**Comment by jehaby** [comment]
maybe it would be better to add `fetch-for-user` method to a session protocol? 
full support of this query language substantially complicates other [possible] session implementation.

**Comment by g-krisztian** [comment]
> maybe it would be better to add `fetch-for-user` method to a session protocol? full support of this query language substantially complicates others [possible] session implementation.

A fetch-for-key still better option. We have no idea about what other developers will store in the session. This is why this should be flexible and general as possible. We have the :users/id key, but others can have different options for it. But I believe we already make it to convention.

**Comment by jehaby** [comment]
i'm having trouble imagining the case when somebody might need the ability to fetch multiple sessions except getting all sessions for a single user. 
for cases when we identify a user not by `:users/id`, but by some other key, we might add a configuration key in a session constructor.

**Comment by g-krisztian** [comment]
I'ts not a must have feature. If you see it's can be a blocker for moving forward we can close this PR without merging it.

**Comment by jehaby** [comment]
yeah, IMO we shouldn't merge in the current state.

**Comment by g-krisztian** [comment]
Closing as not needed.

**Pull request closed without merging**